### PR TITLE
#164139771 User password reset 

### PR DESCRIPTION
**Pull request by danieldare**
**What does this PR do?**
Resets user's password.

**Description of Task to be completed?**
- Send reset link to user's email.

**How should this be manually tested?**
- Checkout `ft-reset-password-164139771`
- Run `npm i`
- Input an email address to send password reset link by sending a post request to`/api/users/reset-password`. 
- The reset link will be sent to the user's email. 
- After clicking the reset link, use the token in the  `req.query` as a bearer token for updating the password by sending a post request to`/api/users/reset-password`.

**Any background context you want to provide?**
N/A

**What are the relevant pivotal tracker stories?**
#164139771

**Screenshots (if appropriate)**

### CODE REVIEW

File: server/controllers/auth-controller.js [hunk 1]
```diff
@@ -157,6 +161,106 @@ class UsersController {
...
+          const { username, id } = user;
+          const token = generateToken({ username, id });
+          mailer
+            .sendResetMail({
+              email,
+              username,
+              resetPasswordLink: generateResetLink(token)
+            })
+            .then(() => {
+              user.save();
```
Comment (andela-pessien) [inline server/controllers/auth-controller.js]: The user hasn't been modified at this point, no?

**Review by andela-pessien — CHANGES_REQUESTED** [review]
Rebase as well please

File: server/controllers/auth-controller.js (hunk 1 above)
Comment (danieldare) [reply server/controllers/auth-controller.js]: it is totally redundant.

**Review by andela-pessien — CHANGES_REQUESTED** [review]
Please address the Hound violations here. And that drop in coverage is far too much to be approved.

File: server/controllers/auth-controller.js (hunk 1 above)
Comment (andela-pessien) [reply server/controllers/auth-controller.js]: So why is it still here?

File: test/users.spec.js [hunk 2]
```diff
@@ -2,9 +2,12 @@ import chai, { expect } from 'chai';
 import chaiHttp from 'chai-http';
 import faker from 'faker';
 import server from '../server';
+import db from '../server/models';
 import { generateToken } from '../server/utils/helpers';
 import generateRandomUser from './mockdata/mock';
 
+const { User } = db;
```
[the remark this answers is not in the document]
Comment (andela-pessien) [reply test/users.spec.js:9]: Address this please @danieldare

[the remark this answers is not in the document]
Comment (danieldare) [reply test/users.spec.js:9]: it is being used. I don't why this houndci-bot is flagging as never used. :(

File: server/controllers/auth-controller.js (hunk 1 above)
Comment (danieldare) [reply server/controllers/auth-controller.js]: not there anymore :)

**Comment by danieldare** [comment]
> Rebase as well please

Rebased!

**Review by andela-pessien — APPROVED** [review]

**Pull request merged**